ethcore/src/spec/spec.rs

```diff
@@ -101,18 +97,38 @@ pub struct Spec {
 	genesis_state: PodState,
 }
 
-#[cfg_attr(feature="dev", allow(wrong_self_convention))] // because to_engine(self) should be to_engine(&self)
-impl Spec {
-	/// Convert this object into a boxed Engine of the right underlying type.
-	// TODO avoid this hard-coded nastiness - use dynamic-linked plugin framework instead.
-	pub fn to_engine(self) -> Result<Box<Engine>, Error> {
-		match self.engine_name.as_ref() {
-			"NullEngine" => Ok(NullEngine::new_boxed(self)),
-			"Ethash" => Ok(ethereum::Ethash::new_boxed(self)),
-			_ => Err(Error::UnknownEngineName(self.engine_name.clone()))
+impl From<ethjson::spec::Spec> for Spec {
+	fn from(s: ethjson::spec::Spec) -> Self {
+		let builtins = s.accounts.builtins().into_iter().map(|p| (p.0.into(), From::from(p.1))).collect();
+		let g = Genesis::from(s.genesis);
+		let seal: GenericSeal = g.seal.into();
+		let params = CommonParams::from(s.params);
+		Spec {
+			name: s.name.into(),
+			params: params.clone(),
+			engine: match s.engine {
```

Comment:
this `match` should be left in its own function; it's currently a stub for a plugin loader and the code should reflect that.